sdk/core/azure-core/src/main/java/com/azure/core/util/ProgressReceiver.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+/**
+ * A {@link ProgressReceiver} is an interface that can be used to report progress on network transfers.
+ * The {@link #reportProgress(long)} method will be called periodically with the total number of bytes
+ * transferred.
+ */
+@FunctionalInterface
+public interface ProgressReceiver {
```

Comment:
Do we have any worries about how generic the class name is? Do we want to make this more specific since it's no longer in a Storage context where other SDKs may have concepts of progress such as for long-running operations, maybe `NetworkProgressReceiver`

Reply:
It can be used to track disk write progress as well. Like here https://github.com/kasobol-msft/azure-sdk-for-java/commit/9f37baef0bbc4da0396af3a5f05f0ead89f11095 .

ProgressReceiver isn't http/network concept by its own. We provide a way to hook it up with network uploads here due to needs, but nothing prevents from using it to track processes like serialization, compression, disk writes and so on.